app/src/@anchor-protocol/app-provider/queries/evm/evmBlockNumber.ts

```diff
@@ -0,0 +1,26 @@
+import { useAnchorWebapp } from '@anchor-protocol/app-provider/contexts/context';
+import { ANCHOR_QUERY_KEY } from '@anchor-protocol/app-provider/env';
+import { useEvmWallet } from '@libs/evm-wallet';
+import { useQuery } from 'react-query';
+
+export function useEvmBlockNumber() {
+  const { provider } = useEvmWallet();
+  const { queryErrorReporter } = useAnchorWebapp();
+
+  const result = useQuery(
+    [ANCHOR_QUERY_KEY.EVM_BLOCK_NUMBER, provider?.network?.chainId],
+    async () => {
+      const blockNumber = await provider?.getBlockNumber();
+      console.log(provider);
+      return blockNumber;
+    },
+    {
+      refetchInterval: 1000 * 60,
+      keepPreviousData: true,
```

Comment:
can set this to false as we dont need to ever fetch old values from this query.

Reply:
✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/304/commits/2ab64bdbb4da953bb9c3ba77cff867a31f94586e